vault/identity_store_entities.go

```diff
@@ -411,6 +411,14 @@ func (i *IdentityStore) pathEntityIDRead(req *logical.Request, d *framework.Fiel
 		return nil, nil
 	}
 
+	return i.handleEntityReadCommon(entity)
+}
+
+func (i *IdentityStore) handleEntityReadCommon(entity *identity.Entity) (*logical.Response, error) {
+	if entity == nil {
+		return nil, fmt.Errorf("nil entity")
```

Comment:
Maybe we should check for `entity == nil` in `pathLookupEntityUpdate` so we can return a better error message, with the right http code

Reply:
Handling entity being nil in `pathLookupEntityUpdate` now.